velox/exec/Window.cpp

```diff
@@ -300,7 +317,29 @@ void Window::updateKRowsFrameBounds(
   if (frameArg.index == kConstantChannel) {
     auto constantOffset = frameArg.constant.value();
     auto startValue =
-        startRow + (isKPreceding ? -constantOffset : constantOffset);
+        (isKPreceding ? -constantOffset : constantOffset) + startRow;
+
+    if (isKPreceding) {
+      if (startValue < INT32_MIN) {
```

Comment:
@aditi-pandit, thanks for your comment!
The handling here is based on an assumption, i.e., any window partition's size cannot be larger than INT32_MAX (I also note `rawFrameBounds` is int32_t*). Thus, `startRow - constantOffset` cannot be a positive number when `constantOffset` is so large. So I directly set 0 for all rows here.

Reply:
Ah I see. Maybe rephrase the comment as follows for better clarity
"For overflow in kPreceding frames, k < INT32_MIN. Since the max number of rows in a partition is INT32_MAX, the frameBound will always be bound to the first row of the partition"

Reply:
@aditi-pandit, yes, your suggested comment is better. Just updated. Thanks!